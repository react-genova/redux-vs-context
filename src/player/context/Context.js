import { createContext } from 'react';
import { createSelector } from 'reselect';

const PlayerContext = createContext({
    name: '',
    surname: '',
    onNameChanged: () => null,
    onSurnameChanged: () => null
});

const getPlayer = value => value;
export const getPlayerName = createSelector(
    getPlayer,
    player => player.name
);
export const getPlayerSurname = createSelector(
    getPlayer,
    player => player.surname
);
export const getPlayerNameChangedCallback = createSelector(
    getPlayer,
    player => player.onNameChanged
);
export const getPlayerSurnameChangedCallback = createSelector(
    getPlayer,
    player => player.onSurnameChanged
);

export default PlayerContext;
