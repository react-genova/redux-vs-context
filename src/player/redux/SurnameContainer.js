import { connect } from 'react-redux';
import FieldShow from '../components/FieldShow';
import { getPlayerSurname } from './player';

export const mapStateToProps = state => ({
    value: getPlayerSurname(state),
    label: 'Surname'
});

const SurnameContainer = connect(mapStateToProps)(FieldShow);

export default SurnameContainer;
