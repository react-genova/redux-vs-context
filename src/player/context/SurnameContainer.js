import React, { memo } from 'react';
import FieldShow from '../components/FieldShow';
import Context from './Context';

const PureFieldShow = memo(FieldShow);

const SurnameContainer = () => (
    <Context.Consumer>
    {({ surname }) => <PureFieldShow value={surname} label="Surname" />}
    </Context.Consumer>
);

export default SurnameContainer;
