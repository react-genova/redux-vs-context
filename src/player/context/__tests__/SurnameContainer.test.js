import React from 'react';
import { render } from 'react-testing-library';
import SurnameContainer from '../SurnameContainer';
import Context, { noop } from '../Context';

jest.mock('../../components/FieldShow', () => ({
    __esModule: true,
    default: ({ value, label }) => (
        <div>
            <div data-testid="surname">{value}</div>
            <div data-testid="label">{label}</div>
        </div>
    )
}));

describe('SurnameContainer consumer container', () => {
    it('renders without crashing and passes values down to its child', () => {
        const { getByTestId, container } = render((
            <Context.Provider
                value={{
                    name: 'Paolo',
                    surname: 'Rossi',
                    onNameChanged: noop,
                    onSurnameChanged: noop,
                }}
            >
                <SurnameContainer />
            </Context.Provider>
        ));
        expect(container.firstChild).toMatchSnapshot();
        expect(getByTestId('surname').textContent).toBe('Rossi');
        expect(getByTestId('label').textContent).toBe('Surname');
    });
});
