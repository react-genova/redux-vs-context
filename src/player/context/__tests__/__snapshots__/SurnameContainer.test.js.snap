// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurnameContainer consumer container renders without crashing and passes values down to its child 1`] = `
<div>
  <div
    data-testid="surname"
  >
    Rossi
  </div>
  <div
    data-testid="label"
  >
    Surname
  </div>
</div>
`;
